### DISCUSSION TITLE
How do I tell which verison I'm on?

### DISCUSSION THREAD

**Discussion by acdoussan** [LazyVim/LazyVim#5227] [General]
https://github.com/LazyVim/LazyVim/blob/main/NEWS.md lists updates quite well, but I have no clue what "version" I'm on, and `<leader>l` doesn't say anything about this either. I'm trying to determine how far back I need to read to determine what I need to change, but I can't find anything about determining what version I'm on.

**Comment by acdoussan** [comment]
Also don't see anything about this in the lazy-lock.json

**Comment by acdoussan** [comment]
I see there is a way to pin the version you are on, did a full search on my config files but I don't have this anywhere.

Maybe it would be wise to include this in the starter project, and periodically update the starter project when new versions are released? This way it is easy to tell?

https://github.com/LazyVim/LazyVim/discussions/5065

**Reply by dpetka2001** [reply]
That's not needed. Lazyvim is just a plugin and gets updated by the package manager like all other plugins. If you press `<leader>l` and then go with your cursor on Lazyvim and press `<CR>` it will show you info about the plugin. And if you press `?` on Lazy UI it will show you the help and you can see what other keybindings exist.

So, no need really to keep any kind of version in the starter since the package manager takes care of everything. Of course, if you want, you can pin to whatever version in your personal configuration. That's why it exists.

**Reply by acdoussan** [reply]
If this is going to make breaking changes (as the past few releases have) I would argue there needs to be an opt-in mechanism to the updates. I shouldn't try to update my packages and end up with a completely broken editor experience because I have personal keybinds for a package that is no longer being used, or because there is a new command line dependency that I don't have on my machine.

**Reply by dpetka2001** [reply]
The opt-in is by reading the breaking changes in the Lazy UI and not press the Update button. Instead you go to your personal configuration and put `{ "LazyVim/LazyVim", version = "13" }` and then you've pinned the version in your personal configuration and it won't show in the updates again until you make the decision. It all comes down to people reading the breaking changes, which the Lazy UI already mentions and not blindly hit the Update button.

**Reply by nipuna-perera** [reply]
Is there a way to rollback if you accidentally updated? I tried setting a different LazyVim version but only the LazyVim plugin rolled back. Not others. That put my config in a bad state naturally.

**Reply by dpetka2001** [reply]
No, every plugin spec is independent and you have to define its corresponding `version`. If you version control your `lazy-lock.json` you could try overwriting the one you have now with a version that you know has the state of the plugins' commit shas that you want and you can open Lazy UI and press `R` to restore the plugins to the ones from your preferred `lazy-lock.json`.

I haven't tried this ever, but from my understanding this is how it should work.

**Reply by acdoussan** [reply]
> The opt-in is by reading the breaking changes in the Lazy UI and not press the Update button.

Sorry, but I think we're just going to have to agree to disagree on this. It's on me if I update my packages and one of them had a breaking change, but IMO there should be at least one more step between users and this proverbial foot gun when the framework as a whole distributes breaking changes.

**Reply by dpetka2001** [reply]
Exactly, we'll just have to agree to disagree. 

In the end it's on the user if he wants to update or not. There are alternatives, like pinning the LazyVim version so the user doesn't update or if he decides to go through with the update he can always enable the corresponding Extra that he wants to keep using instead of the new default ones. The tools for providing the choice to either pin a version or revert to older commits or enable the new Extra to keep using the old defaults are still there and it's a matter of 2-3 key presses to do what he wants. 

The breaking changes are mentioned right on the top before the update and the user can go on with the best option for him by choosing one of the aforementioned methods. If you don't want `fzf-lua` you just enable the Telescope Extra and you're back to your previous workflow.
